src/server/config/schema.js

```diff
@@ -53,6 +53,7 @@ export default () => Joi.object({
     autoListen: Joi.boolean().default(true),
     defaultRoute: Joi.string().default('/app/kibana').regex(/^\//, `start with a slash`),
     basePath: Joi.string().default('').allow('').regex(/(^$|^\/.*[^\/]$)/, `start with a slash, don't end with one`),
+    customResponseHeaders: Joi.object().default({}),
```

Comment:
Looks like omitting `.keys()` here allows the object to have any keys, but I think it would be nice to have an explicit `.unknown(true)` to be extra clear about that intention.